composables/useVerifyAccount.ts

```diff
@@ -30,36 +33,50 @@ export default function useVerifyAccount() {
     return signedMessage
   }
 
-  const getSignedMessage = async () => {
+  const getSignedMessage = async (message) => {
     if (!accountId.value) {
       throw new Error('Please connect your wallet first')
     }
-    if (signedMessage.value) {
-      return signedMessage.value
-    }
 
     const signMessageFn = isEthereumAddress(accountId.value)
       ? signMessageEthereum
       : signMessagePolkadot
-    const signature = await signMessageFn(accountId.value, SIGNATURE_MESSAGE)
+    const signature = await signMessageFn(accountId.value, message)
 
     if (signature) {
-      walletStore.setSignedMessage(signature)
       return signature
     }
 
     throw new Error('You have not completed address verification')
   }
 
-  const getSignaturePair = async () => {
-    const signature = await getSignedMessage()
+  const getCustomSignaturePair = async (message) => {
+    const signature = await getSignedMessage(message)
+    return {
+      signature,
+      message,
+    }
+  }
+
+  const getCommonSignaturePair = async () => {
+    if (signedMessage.value) {
+      return {
+        signature: signedMessage.value,
+        message: SIGNATURE_MESSAGE,
+      }
+    }
+    const signature = await getSignedMessage(SIGNATURE_MESSAGE)
+    if (signature) {
+      walletStore.setSignedMessage(signature)
+    }
```

Comment:
I don't think `signature` can be null here, but if that is the case

```suggestion
    const signature = await getSignedMessage(SIGNATURE_MESSAGE)
    if (!signature) {
      return null // ? or undefined 
    }
    
    walletStore.setSignedMessage(signature)
```

Reply:
If the user cancels the wallet signature process, it will throw error inside `getSignedMessage`. Therefore there's not need to throw an error again.